src/features/defi/components/Deposit/PairDepositWithAllocation.tsx

```diff
@@ -0,0 +1,320 @@
+import get from 'lodash/get'
+import { calculateYearlyYield } from 'plugins/cosmos/components/modals/Staking/StakingCommon'
+import type { ControllerProps } from 'react-hook-form'
+import { useController, useForm, useWatch } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import type { AccountDropdownProps } from 'components/AccountDropdown/AccountDropdown'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { Allocation } from 'components/DeFi/components/Allocation'
+import { AssetInput } from 'components/DeFi/components/AssetInput'
+import { FormField } from 'components/DeFi/components/FormField'
+import { Row } from 'components/Row/Row'
+import { Text } from 'components/Text'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+
+import { PairIcons } from '../PairIcons/PairIcons'
+
+type DepositProps = {
+  accountId?: AccountId | undefined
+  asset1: Asset
+  asset2: Asset
+  destAsset: Asset
+  // Estimated apy (Deposit Only)
+  apy: string
+  // Use form values to calculate allocation fraction
+  calculateAllocations?(
+    asset: Asset,
+    amount: string,
+  ): { allocationFraction: string; shareOutAmount: string }
+  // Users available amount
+  cryptoAmountAvailable1: string
+  cryptoAmountAvailable2: string
+  // Validation rules for the crypto input
+  cryptoInputValidation1?: ControllerProps['rules']
+  cryptoInputValidation2?: ControllerProps['rules']
+  // Users available amount
+  fiatAmountAvailable1: string
+  fiatAmountAvailable2: string
+  // Validation rules for the fiat input
+  fiatInputValidation1?: ControllerProps['rules']
+  fiatInputValidation2?: ControllerProps['rules']
+  // enables slippage UI (defaults to true)
+  enableSlippage?: boolean
+  // Asset1 market data
+  marketData1: MarketData
+  // Asset2 market data
+  marketData2: MarketData
+  // Array of the % options
+  percentOptions: number[]
+  isLoading: boolean
+  onAccountIdChange?: AccountDropdownProps['onChange']
+  onContinue(values: DepositValues): void
+  onBack?(): void
+  onCancel(): void
+  syncPair?: boolean
+  icons?: string[]
+}
+
+enum Field {
+  FiatAmount1 = 'fiatAmount1',
+  FiatAmount2 = 'fiatAmount2',
+  CryptoAmount1 = 'cryptoAmount1',
+  CryptoAmount2 = 'cryptoAmount2',
+  AllocationFraction = 'allocationFraction',
+  ShareOutAmount = 'shareOutAmount',
+}
+
+export type DepositValues = {
+  [Field.FiatAmount1]: string
+  [Field.CryptoAmount1]: string
+  [Field.FiatAmount2]: string
+  [Field.CryptoAmount2]: string
+  [Field.AllocationFraction]: string
+  [Field.ShareOutAmount]: string
+}
+
+export const PairDepositWithAllocation = ({
+  accountId,
+  apy,
+  asset1,
+  asset2,
+  calculateAllocations,
+  cryptoAmountAvailable1,
+  cryptoAmountAvailable2,
+  cryptoInputValidation1,
+  cryptoInputValidation2,
+  destAsset,
+  fiatAmountAvailable1,
+  fiatAmountAvailable2,
+  fiatInputValidation1,
+  fiatInputValidation2,
+  icons,
```

Comment:
This can also be selected, since this is coming from `opportunity?.icons`, will make the component API not lightweight by selecting it in-place vs. passing it down